Sources/Utility/ArgumentParser.swift

```diff
@@ -294,16 +281,25 @@ final class AnyArgument: ArgumentProtocol, CustomStringConvertible {
 /// Argument parser protocol passed in initializers of ArgumentKind to manipulate
 /// parser as needed by the argument.
 public protocol ArgumentParserProtocol {
-    /// The current argument being parsed.
-    var currentArgument: String { get }
-
     /// The associated value in a `--foo=bar` style argument.
     var associatedArgumentValue: String? { get }
 
     /// Provides (and consumes) next argument, if available.
     mutating func next() throws -> String
 }
 
+/// A protocol representing a parser option or positional argument.
+protocol ParserArgument {
+    /// Backed argument.
+    var argument: AnyArgument { get }
+    /// Array parsing strategy to use.
+    var strategy: ArgumentParser.ArrayParsingStrategy { get }
```

Comment:
Okay. We'd have to store the `ArrayParsingStrategy` in `ArgumentProtocol` then. You okay with that?